### PR TITLE
Inclusive Code Reviews

### PR DESCRIPTION
**Pull request by lakshaykaushik**
# Pull Request Template

## What are you trying to address

- Describe the current behavior that you are modifying, and link to issue number
- If you don't have an issue, browse through existing issues to see if this is already tracked as an issue, to assign yourself to the issue and also verify that no one else is already working on the issue.

## Description of new changes

- Write a detailed description of all changes - and if appropriate, why they are needed

## For all pull requests

- [ ] Link to the issue you are solving (so it gets closed)
- [ ] Label the pull request with the appropriate area(s)
- [ ] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [ ] Assign the project - Engineering Playbook Backlog
- [ ] Assign the pull request to yourself

### CODE REVIEW

File: docs/code-reviews/inclusion-in-code-review.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+* Inequitable review assignment;
+* Negative Inter-Personal Interactions;
+* Biased Decision Making;
+
+## Examples of Non-Inclusive Code Reviews
+
+* Assigning lot of PRs to a single person for review and not assigning any PRs for other people in team. - _Inequitable review assignment_
+* Long arguments in the pull requests over subjective topics like style of coding . - _Negative Inter-Personal Interactions_
+* Commenting on the developer and not the code. Assuming code from developer X will always be good and hence not reviewing it properly and vice versa. - _Biased Decision Making_
+
```
[the remark this answers is not in the document]
Comment (tompaana) [reply docs/code-reviews/inclusion-in-code-review.md:22]: My suggestions merge this with the previous section.

File: docs/code-reviews/inclusion-in-code-review.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+## Examples of Inclusive Code Reviews
+
+* Anyone and everyone in the team should be assigned PRs to review.
+* Reviewer should be clear about what is an opinion, their personal preference, best practice or a fact. Arguments over personal preferences and opinions are mostly avoidable.
+* Using inclusive language and tone in the code review comments. For example, being suggestive rather being prescriptive in the review comments is a good way to get the point across the table.
+* For the author of PRs- It's a good practice to thank the reviewer for the review if they have contributed in improving your code or you have learnt something new.
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:28]:
```suggestion
* It's a good practice for the author of a PR to thank the reviewer for the review, when they have contributed in improving the code or you have learnt something new.
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+* Using the Sandwich technique, for recommending a code change to a new developer or a new customer, sandwich the suggestion between 2 compliments. Something like- "Great Work so far, but i would recommend a few changes here. Btw, i loved the use of XYZ here, great work!!."
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:29]:
```suggestion
* Using the sandwich method for recommending a code change to a new developer or a new customer: Sandwich the suggestion between 2 compliments. For example: "Great work so far, but I would recommend a few changes here. Btw, I loved the use of XYZ here, nice job!"
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+## Responsibility of the author to make code reviews inclusive
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:31]:
```suggestion
## Guidelines for the Author
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+* Aim to write a code that is easy to read, review and maintain.
+* It’s important to ensure that whoever is looking at the code, whether that be the reviewer or a future engineer, can understand the motivations and how your code achieves that.
+* Proactively asking for targeted help or feedback.
+* Respond clearly to questions asked by the reviewers.
+* Avoid huge commits to submitting incremental changes.
+
+## Responsibility of the Reviewer to make code reviews inclusive
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:39]:
```suggestion
## Guidelines for the Reviewer
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+* Anyone and everyone in the team should be assigned PRs to review.
+* Reviewer should be clear about what is an opinion, their personal preference, best practice or a fact. Arguments over personal preferences and opinions are mostly avoidable.
+* Using inclusive language and tone in the code review comments. For example, being suggestive rather being prescriptive in the review comments is a good way to get the point across the table.
+* For the author of PRs- It's a good practice to thank the reviewer for the review if they have contributed in improving your code or you have learnt something new.
+* Using the Sandwich technique, for recommending a code change to a new developer or a new customer, sandwich the suggestion between 2 compliments. Something like- "Great Work so far, but i would recommend a few changes here. Btw, i loved the use of XYZ here, great work!!."
+
+## Responsibility of the author to make code reviews inclusive
+
+* Aim to write a code that is easy to read, review and maintain.
+* It’s important to ensure that whoever is looking at the code, whether that be the reviewer or a future engineer, can understand the motivations and how your code achieves that.
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:34]:
```suggestion
* It’s important to ensure that whoever is looking at the code, whether that be the reviewer or a future engineer, can understand the motivations and how your code achieves its goals.
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+* Proactively asking for targeted help or feedback.
+* Respond clearly to questions asked by the reviewers.
+* Avoid huge commits to submitting incremental changes.
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:37]: "Avoid huge commits to submitting incremental changes." - I'm not sure I understand exactly what is meant by this bullet point. Is it that one should favor small incremental changes over overhauls? I think this goes more into the pull request category and furthermore, how user stories are divided into tasks and their acceptance criteria etc. etc.

I suggest removing this line or rephrasing it to reveal the inclusiveness aspect of it.

File: docs/code-reviews/inclusion-in-code-review.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+## Responsibility of the Reviewer to make code reviews inclusive
+
+* Assume positive intent from the author
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:41]:
```suggestion
* Assume positive intent from the author.
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+* Write clear and elaborative comments on the PR
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:42]:
```suggestion
* Write clear and elaborate comments.
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+* Identify subjectivity, choice of coding and best practice. It is good to discuss coding style and subjective coding choices in some other forum and not in the PR. A PR should not become a ground to discuss subjective coding choices and having long arguments over it
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:43]:
```suggestion
* Identify subjectivity, choice of coding and best practice. It is good to discuss coding style and subjective coding choices in some other forum and not in the PR. A PR should not become a ground to discuss subjective coding choices and having long arguments over it.
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+* If you do not understand the code properly, refrain from commenting like for e.g. "This code is incomprehensible". Better is to get in a call with the author and get a basic understanding of what they have done.
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:44]:
```suggestion
* If you do not understand the code properly, refrain from commenting e.g., "This code is incomprehensible". It is better to have a call with the author and get a basic understanding of their work.
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+In such cases, assuming positive intent of the author and reviewer is a good start to start analyzing quality of code reviews.
+
+## Tools
+
+Below are some tools which may help in establishing inclusive code review culture within our teams.
+
+* [anonymous_github](https://github.com/tdurieux/anonymous_github)
+* [blind-reviews](https://github.com/zombie/blind-reviews/)
+* [gitmask](https://www.gitmask.com/)
+* [inclusivelint](https://github.com/inclusivelint)
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:59]:
```suggestion
* [Anonymous Github](https://github.com/tdurieux/anonymous_github)
* [Blind Code Reviews](https://github.com/zombie/blind-reviews)
* [Gitmask](https://www.gitmask.com)
* [inclusivelint](https://github.com/inclusivelint)
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+* Assume positive intent from the author
+* Write clear and elaborative comments on the PR
+* Identify subjectivity, choice of coding and best practice. It is good to discuss coding style and subjective coding choices in some other forum and not in the PR. A PR should not become a ground to discuss subjective coding choices and having long arguments over it
+* If you do not understand the code properly, refrain from commenting like for e.g. "This code is incomprehensible". Better is to get in a call with the author and get a basic understanding of what they have done.
+* Be suggestive and not prescriptive. A reviewer should suggest changes and not prescribe changes, let the author decide if they really want to accept the changes proposed.
+
+## Culture and Code Reviews
+
+We as CSE, may come across situations in which code reviews are not ideal and often we are observing non inclusive code review behaviors. Its important to be cognizant of the fact that culture and communication style of a particular geography also influences how people interact over pull requests.
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:49]:
```suggestion
We as CSE, may come across situations in which code reviews are not ideal and often we are observing non inclusive code review behaviors. Its important to be aware of the fact that culture and communication style of a particular geography also influences how people interact over pull requests.
```

File: docs/code-reviews/inclusion-in-code-review.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
+# Inclusion in Code Review
+
+You may be thinking why care about inclusiveness in code reviews. Below are some points which emphasize why inclusivity in code reviews is important:
```
Comment (tompaana) [inline docs/code-reviews/inclusion-in-code-review.md:3]: We could also be direct here e.g., "Inclusiveness in code review is important because:"

**Review by tompaana — CHANGES_REQUESTED** [review]
Excellent content! I added bunch of suggestions to help clean it a bit and to make the style more uniform. Bulleted lists are my nemeses so I resorted to this guide I found online: [How to Correctly Punctuate Bullet Point Lists](https://www.instructionalsolutions.com/blog/bulleted-list-punctuation). Please commit the suggestions you like and ping me again so I can approve. Thanks!

File: docs/code-reviews/inclusion-in-code-review.md (hunk 14 above)
Comment (lakshaykaushik) [reply docs/code-reviews/inclusion-in-code-review.md]: done

**Comment by lakshaykaushik** [comment]
@tompaana thanks for the awesome review. I have incorporated the changes. Can you review and approve.

**Review by tompaana — APPROVED** [review]
Approving this fine PR.

**Review by TessFerrandez — APPROVED** [review]
Great advice

**Pull request merged**